ably/websocket.go

```diff
@@ -88,30 +103,30 @@ func dialWebsocket(proto string, u *url.URL, timeout time.Duration, agents map[s
 		return nil, errors.New(`invalid protocol "` + proto + `"`)
 	}
 	// Starts a raw websocket connection with server
-	conn, err := dialWebsocketTimeout(u.String(), "https://"+u.Host, timeout, agents)
+	conn, resp, err := dialWebsocketTimeout(u.String(), "https://"+u.Host, timeout, agents)
 	if err != nil {
-		return nil, err
+		return nil, &websocketErr{err: err, resp: resp}
 	}
```

Comment:
While creating `websocketErr` object, we already make sure `err` is non-nil. So, there's no point in checking internally if `err` is nil or not. This also means rabbitAi posted some review suggestions for non-required checks.